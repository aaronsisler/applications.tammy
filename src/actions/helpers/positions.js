export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
