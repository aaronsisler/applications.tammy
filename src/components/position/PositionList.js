import React from 'react';
import PropTypes from 'prop-types';
import PositionListItem from './PositionListItem';

export default class PositionList extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
    }

    render() {
        if (this.props.positions.length === 0) {
            return (
                <div className="position_list empty">
                    No available items
                </div>
            );
        }

        return (
            <div className="position_list">
                {
                    this.props.positions.map((position) =>
                        <PositionListItem
                            key={position.positionId}
                            {...position}
                        />)
                }
            </div>
        );
    }
}

PositionList.propTypes = {
    positions: PropTypes.array,
};

