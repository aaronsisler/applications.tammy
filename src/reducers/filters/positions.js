const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
