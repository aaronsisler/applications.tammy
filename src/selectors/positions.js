// Get filtered positions
export default (positions, { text = '' }) => positions
        .filter((position) => {
            const titleMatch = position.title.toLowerCase().includes(text.toLowerCase());
            const jobIdMatch = position.jobId.toLowerCase().includes(text.toLowerCase());
            return jobIdMatch || titleMatch;
        })
