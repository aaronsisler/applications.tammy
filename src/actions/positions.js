import database from '../firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setPositions(positions));
    });
