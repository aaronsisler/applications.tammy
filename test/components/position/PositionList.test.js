import React from 'react';
import { shallow } from 'enzyme';
import PositionList from 'Position/PositionList';
import positions from '../../fixtures/positions';

describe('PositionList', () => {
    let wrapper;

    const buildWrapper = (positionsInput = []) => {
        wrapper = shallow(
            <PositionList
                positions={positionsInput}
            />);
    }

    it('should render PositionList correctly when positions are available', () => {
        buildWrapper(positions);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render PositionList correctly when no positions are available', () => {
        buildWrapper();
        expect(wrapper).toMatchSnapshot();
    });
});
