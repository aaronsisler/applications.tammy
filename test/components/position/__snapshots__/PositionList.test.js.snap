// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionList should render PositionList correctly when no positions are available 1`] = `
<div
  className="position_list empty"
>
  No available items
</div>
`;

exports[`PositionList should render PositionList correctly when positions are available 1`] = `
<div
  className="position_list"
>
  <Connect(PositionListItem)
    content={
      Object {
        "description": "first description",
        "requirements": Array [
          "Skill 1",
          "Skill 2",
        ],
        "responsibilities": Array [
          "Responsibility 1",
          "Responsibility 2",
        ],
      }
    }
    jobId="first job id"
    key="firstPositionId"
    location="first location"
    positionId="firstPositionId"
    title="first title"
  />
  <Connect(PositionListItem)
    content={
      Object {
        "description": "second description",
        "requirements": Array [
          "Skill 1",
          "Skill 2",
        ],
        "responsibilities": Array [
          "Responsibility 1",
          "Responsibility 2",
        ],
      }
    }
    jobId="second job id"
    key="secondPositionId"
    location="second location"
    positionId="secondPositionId"
    title="second title"
  />
</div>
`;
